app/client/components/DeleteModeButton.js

```diff
@@ -0,0 +1,33 @@
+
+import React from 'react';
+import Glyphicon from 'react-bootstrap/lib/Glyphicon';
+
+export default class DeleteModeButton extends React.Component {
+
+    static propTypes = {
+        isActive: React.PropTypes.bool,
+        onClick: React.PropTypes.func,
+    }
+
+    render() {
+        var { isActive } = this.props;
+        var { handleChange } = this;
+
+        var deleteMode = isActive ? 'shake-little' : 'dont-shake';
+
+        return (
+            <div onClick={handleChange} className={deleteMode}>
+                <Glyphicon
+                    glyph={'glyphicon glyphicon-trash'}
+                    onClick={handleChange} className={deleteMode} />
```

Comment:
the method `handleChange` is assigned to both the icon and the wrapping DIV.
This is not necessary.

I believe we can get rid of the DIV wrapper and apply the class name directly to the icon.